rotkehlchen/api/rest.py

```diff
@@ -4155,3 +4082,125 @@ def add_evm_transaction_by_hash(
+            query_filter: HistoryEventFilterQuery,
+            value_filter: HistoryEventFilterQuery,
+            event_subtypes: Optional[list[HistoryEventSubType]] = None,
+            exclude_subtypes: Optional[list[HistoryEventSubType]] = None,
+    ) -> dict[str, Any]:
+        """Query exchanges for either staking or savings history.
+
+        If `only_cache` is False, only data stored in the database is returned. Otherwise,
+        the latest data is fetched from the exchange.
+        """
+        history_events_db = DBHistoryEvents(self.rotkehlchen.data.db)
+        table_filter = HistoryEventFilterQuery.make(
+            location=location,
+            event_types=event_types,
+            event_subtypes=event_subtypes,
+            exclude_subtypes=exclude_subtypes,
+        )
+
+        message = ''
+        entries_limit = -1
+        if self.rotkehlchen.premium is None:
+            entries_limit = FREE_HISTORY_EVENTS_LIMIT
+
+        exchanges_list = self.rotkehlchen.exchange_manager.connected_exchanges.get(
+            location,
+        )
+        if exchanges_list is None:
+            return wrap_in_fail_result(
+                message=f'There is no {location.name} account added.',
+                status_code=HTTPStatus.CONFLICT,
+            )
+
+        # After 3865 we have a recurring task that queries for missing prices but
+        # we make sure that the returned values have their correct value calculated
+        task_manager = self.rotkehlchen.task_manager
+        if task_manager is not None:
+            try:
+                entries = task_manager.get_base_entries_missing_prices(query_filter)
+                task_manager.query_missing_prices_of_base_entries(
+                    entries_missing_prices=entries,
+                )
+            except sqlcipher.OperationalError as e:  # pylint: disable=no-member
+                return wrap_in_fail_result(
+                    message=f'Database query error retrieving misssing prices {str(e)}',
+                    status_code=HTTPStatus.CONFLICT,
+                )
+
+        # Query events from database
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            events_raw, entries_found = self.rotkehlchen.events_historian.query_history_events(
+                cursor=cursor,
+                location=location,
+                filter_query=query_filter,
+                only_cache=only_cache,
+            )
+            events: list[Any] = []
+            if location == Location.KRAKEN:
+                for event in events_raw:
+                    try:
+                        staking_event = StakingEvent.from_history_base_entry(event)
+                    except DeserializationError as e:
+                        log.warning(f'Could not deserialize staking event: {event} due to {str(e)}')  # noqa: E501
+                        continue
+                    events.append(staking_event)
+            else:
```

Comment:
Why not count them ast staking events here? It's the same thing. Just keep it simple and a single interface